<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BPControl</class>
 <widget class="QWidget" name="BPControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>578</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="algoGroupBox">
     <property name="title">
      <string>Algorithm</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QRadioButton" name="maxProdButton">
        <property name="text">
         <string>Max-Product</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="sumProdButton">
        <property name="text">
         <string>Sum-Product</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="updatesGroupBox">
     <property name="title">
      <string>Updates</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QRadioButton" name="seqrndButton">
        <property name="text">
         <string>Sequential random</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="seqmaxButton">
        <property name="text">
         <string>Sequential max-residual</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="seqfixButton">
        <property name="text">
         <string>Sequential fixed</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="parallButton">
        <property name="text">
         <string>Parallel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="paramLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="maxIterTextLabel">
       <property name="text">
        <string>Maximum iterations:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="maxIterSpinBox">
       <property name="maximum">
        <number>1000000</number>
       </property>
       <property name="value">
        <number>200</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="tolTextlabel">
       <property name="text">
        <string>Tolerance:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QDoubleSpinBox" name="tolSpinBox">
       <property name="decimals">
        <number>10</number>
       </property>
       <property name="maximum">
        <double>1000000.000000000000000</double>
       </property>
       <property name="value">
        <double>0.000001000000000</double>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="iterTextLabel">
       <property name="text">
        <string>Iterations:</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QSpinBox" name="iterSpinBox">
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="maximum">
        <number>1000000</number>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QCheckBox" name="logDomainCheckBox">
       <property name="text">
        <string>Log-domain</string>
       </property>
      </widget>
     </item>
     <item row="9" column="0">
      <widget class="QLabel" name="llLabel">
       <property name="text">
        <string>Log-likelihood:</string>
       </property>
      </widget>
     </item>
     <item row="9" column="1">
      <widget class="QDoubleSpinBox" name="llSpinBox">
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="decimals">
        <number>10</number>
       </property>
       <property name="maximum">
        <double>1000000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="10" column="0">
      <widget class="QLabel" name="timeLabel">
       <property name="text">
        <string>Time [s]:</string>
       </property>
      </widget>
     </item>
     <item row="10" column="1">
      <widget class="QDoubleSpinBox" name="timeSpinBox">
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="decimals">
        <number>10</number>
       </property>
       <property name="maximum">
        <double>1000000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="strengthLabel">
       <property name="text">
        <string>Strength:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QDoubleSpinBox" name="strengthSpinBox">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="maximum">
        <double>1000000.000000000000000</double>
       </property>
       <property name="value">
        <double>10.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="runLayout">
     <item row="0" column="0">
      <widget class="QPushButton" name="runButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Run</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="Line" name="line_8">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="showLayout">
     <item row="0" column="0">
      <widget class="QCheckBox" name="showBPCheckBox">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Show BP segmentation</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
  <action name="testAction">
   <property name="text">
    <string>Test</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>showBPCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>BPControl</receiver>
   <slot>showBPToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxIterSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BPControl</receiver>
   <slot>maxIterChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tolSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>BPControl</receiver>
   <slot>tolChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runButton</sender>
   <signal>pressed()</signal>
   <receiver>BPControl</receiver>
   <slot>runPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>logDomainCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>BPControl</receiver>
   <slot>logDomainToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>seqfixButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>BPControl</receiver>
   <slot>seqfixToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>seqmaxButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>BPControl</receiver>
   <slot>seqmaxToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>seqrndButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>BPControl</receiver>
   <slot>seqrndToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>170</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>parallButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>BPControl</receiver>
   <slot>parallToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>114</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sumProdButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>BPControl</receiver>
   <slot>sumProdToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxProdButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>BPControl</receiver>
   <slot>maxProdToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>strengthSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>BPControl</receiver>
   <slot>strengthChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showBPToggled(bool)</slot>
  <slot>maxIterChanged(int)</slot>
  <slot>tolChanged(double)</slot>
  <slot>runPressed()</slot>
  <slot>logDomainToggled(bool)</slot>
  <slot>parallToggled(bool)</slot>
  <slot>seqfixToggled(bool)</slot>
  <slot>seqrndToggled(bool)</slot>
  <slot>seqmaxToggled(bool)</slot>
  <slot>maxProdToggled(bool)</slot>
  <slot>sumProdToggled(bool)</slot>
  <slot>strengthChanged(double)</slot>
 </slots>
</ui>
